user_tools/src/spark_rapids_pytools/cloud_api/dataproc.py

```diff
@@ -593,50 +600,39 @@ def get_cluster_configuration(self) -> dict:
         Overrides to provide the cluster configuration which is specific to Dataproc.
         """
         cluster_config = super().get_cluster_configuration()
-        gpu_per_machine, gpu_device = self.get_gpu_per_worker()
-        # Need to handle case this was CPU event log and just make a recommendation
-        gpu_device_hash = {
-            'T4': 'nvidia-tesla-t4',
-            'L4': 'nvidia-l4'
-        }
-        if gpu_device and gpu_per_machine > 0:
-            additional_config = {
-                'gpuInfo': {
-                    'device': gpu_device_hash.get(gpu_device),
-                    'gpuPerWorker': gpu_per_machine
-                },
-                'additionalConfig': {
-                    'localSsd': 2
-                }
-            }
-            cluster_config.update(additional_config)
-        elif gpu_per_machine == 0:
-            # TODO - we should make this smarter about gpuPerWorker
-            # recommended device should match the scala code for Dataproc platform
-            recommended_device = 'nvidia-tesla-t4'
-            if gpu_device:
-                recommended_device = gpu_device_hash.get(gpu_device)
-
-            additional_config = {
-                'gpuInfo': {
-                    'device': recommended_device,
-                    'gpuPerWorker': 1
-                },
-                'additionalConfig': {
-                    'localSsd': 2
-                }
-            }
-            cluster_config.update(additional_config)
-
+        # If the cluster is GPU cluster, we need to add the GPU configuration
+        if self.is_gpu_cluster():
+            gpu_config = self._get_gpu_configuration()
+            cluster_config.update(gpu_config)
+            ssd_config = self._get_ssd_configuration()
+            cluster_config.update(ssd_config)
         return cluster_config
 
+    @classmethod
+    def _get_ssd_configuration(cls) -> dict:
+        """
+        TODO: We should recommend correct number of SSDs instead a fixed number.
+        """
+        return {
+            'ssdInfo': {
+                'numLocalSsds': 2
```

Comment:
nit: I am not sure if it is more clear to say `SSDsPerWorker`, but either way is fine to me

Reply:
That makes sense. Updated the label.